component/config.go

```diff
@@ -110,6 +112,46 @@ func callValidateIfPossible(v reflect.Value) error {
 // Type is the component type as it is used in the config.
 type Type string
 
+var _ fmt.Stringer = (Type)("")
+
+// String returns the string representation of the type.
+func (t Type) String() string {
+	return string(t)
+}
+
+// typeRegexp is used to validate the type of a component.
+// A type must start with an ASCII alphabetic character and
```

Comment:
I dont think there are any examples in Contrib, but is there any fixed reason why someone couldnt start a component type with a number?

Reply:
One argument in favor of not allowing numbers is that it makes the component types conform to the [Unicode default identifier syntax](https://unicode.org/reports/tr31/#D1). If we ever want to expand from ASCII to the full set of identifier allowed by Unicode, we should restrict it now.

Another argument is just that we should start being conservative about the set of identifiers we allow. Restricting the set of identifiers would be a breaking change, expanding the set of identifiers is not a breaking change. So if in doubt, we should leave it out for now and think about it later.

Reply:
You've convinced me